xmtp/src/client.rs

```diff
@@ -105,13 +105,37 @@ where
 
         let mut contacts = vec![];
         for envelope in response.envelopes {
-            let contact_bundle = Contact::from_bytes(envelope.message, wallet_address.to_string())?;
-            contacts.push(contact_bundle);
+            let contact_bundle = Contact::from_bytes(envelope.message)?;
+            match contact_bundle.association() {
```

Comment:
It would be preferred if the wallet check doesn't have to be impolemented everytime the contact is consumed. 

Passing the expected address into `.association` or `::from_bytes` would mean that callers would be forced to verify that the association is valid for this address, before they could access the bundle. 

Is something like that possible?

Reply:
We can include it as an optional argument for cases where we know the wallet address in advance (reading contacts). 

For cases where we don't know the wallet address (reading invites) and are relying on other validations (authenticated encryption in the ciphertext) we would need to be able to omit that check

Reply:
Gave adding an optional `expected_wallet_address` field a shot here: https://github.com/xmtp/libxmtp/pull/139/commits/452402c574e858dea94ab0bedac501267202f05e